tests/unit_tests/test_rating_service.py

```diff
@@ -162,6 +162,12 @@ async def test_load_rating_type_ids(uninitialized_service):
         "ladder_1v1": 2,
         "tmm_2v2": 3
     }
+    global_ = Leaderboard(1, "global")
+    assert service.leaderboards == {
+        "global": global_,
+        "ladder_1v1": Leaderboard(2, "ladder_1v1"),
+        "tmm_2v2": Leaderboard(3, "tmm_2v2", global_)
```

Comment:
This seems to ignore the initializer leaderboard. No matter what I input in the test data, it always passes.

Reply:
Good catch, looks like the generated implementation for `__eq__` was ignoring the leaderboard. Adding a type annotation fixes it